### PR TITLE
Definitions for OpenUI5

### PR DESCRIPTION
**Pull request by maylukas**
Please fill in this template.

- [x] Prefer to make your PR against the `types-2.0` branch.
- [ ] The package does not provide its own types, and you can not add them.
- [x] Test the change in your own code.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes).

If adding a new definition:
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Run `tsc` without errors.
- [x] Include the required [files](https://github.com/DefinitelyTyped/DefinitelyTyped#create-a-new-package) and header.

If changing an existing definition:
- [ ] Provide a URL to  documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] Increase the version number in the header if appropriate.

Added definitions for SAPUI5 / OpenUI5 by parsing the documentation provided by SAP

### CODE REVIEW

File: openui5/jquery.sap.d.ts [hunk 1]
```diff
@@ -0,0 +1,187 @@
+// Type definitions for OpenUI5
+// Project: http://openui5.org/
+// Definitions by: Lukas May <https://www.dscsag.com/>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped 
+
+/// <reference path="../jquery/index.d.ts" />
```
Comment (andy-ms) [inline openui5/jquery.sap.d.ts]: Prefer to use `<reference types="jquery"/>`.

File: openui5/jquery.sap.d.ts [hunk 2]
```diff
@@ -0,0 +1,187 @@
+// Type definitions for OpenUI5
```
Comment (andy-ms) [inline openui5/jquery.sap.d.ts]: Looks like the latest stable release is `1.40`. So write that as `// Type definitions for OpenUI5 1.40`.

File: openui5/ui5-tests.ts [hunk 3]
```diff
@@ -0,0 +1,111 @@
+/// <reference path="sap.m.d.ts" />
```
Comment (andy-ms) [inline openui5/ui5-tests.ts]: Don't need the reference, `tsconfig` should provide it.

File: openui5/tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,17 @@
+{
+    "compilerOptions": {
+        "module": "commonjs",
+        "target": "es6",
+        "noImplicitAny": false,
```
Comment (andy-ms) [inline openui5/tsconfig.json]: Prefer to make at least `noImplicitAny` true. We normally prefer `strictNullChecks` too, but if the documentation you're parsing doesn't have nullability information, that's OK.

File: openui5/tsconfig.json [hunk 5]
```diff
@@ -2,16 +2,10 @@
     "compilerOptions": {
         "module": "commonjs",
         "target": "es6",
-        "noImplicitAny": false,
+        "noImplicitAny": true,
         "strictNullChecks": false,
-        "types": [],
+        "types": ["jquery"],
```
Comment (andy-ms) [inline openui5/tsconfig.json]: The `/// <reference path="../jquery/index.d.ts" />` line should still be removed.

File: openui5/tsconfig.json [hunk 6]
```diff
@@ -2,16 +2,10 @@
...
         "noEmit": true,
         "forceConsistentCasingInFileNames": true
-    },
-    "files": [
```
Comment (andy-ms) [inline openui5/tsconfig.json]: Why did you remove this?

File: openui5/tsconfig.json (hunk 5 above)
Comment (maylukas) [inline openui5/tsconfig.json]: If I remove the reference it gives me an the error "error TS2688: Cannot find type definition file for 'jquery'.". How can I resolve this?

File: openui5/tsconfig.json (hunk 6 above)
Comment (maylukas) [inline openui5/tsconfig.json]: That was actually a mistake. I will re-add it in the next commit

File: openui5/tsconfig.json (hunk 5 above)
Comment (andy-ms) [inline openui5/tsconfig.json]: Use this:

```json
{
    "compilerOptions": {
        "module": "commonjs",
        "target": "es6",
        "noImplicitAny": true,
        "strictNullChecks": false,
        "baseUrl": "../",
        "typeRoots": [
            "../"
        ],
        "types": [],
        "noEmit": true,
        "forceConsistentCasingInFileNames": true
    },
    "files": [
        "jquery.sap.d.ts",
        "sap.m.d.ts",
        "sap.ui.d.ts",
        "ui5-tests.ts"
    ]
}
```

And at the top of `jquery.sap.d.ts` (after the header), write `/// <reference types="jquery" />`.

On another note, this package has no `index.d.ts`. So it must be used like `/// <reference types="openui5/sap.ui" />`.
If there were an `index.d.ts`, it could be automatically referenced.
If people are likely to need both `sap.m` and `sap.ui`, there should be an index referencing both of those (with `/// <reference path="./sap.m.d.ts" />`).
Also, only the index will need a header.

File: openui5/jquery.sap.d.ts [hunk 7]
```diff
@@ -0,0 +1,187 @@
+// Type definitions for OpenUI5 1.40.
```
Comment (andy-ms) [inline openui5/jquery.sap.d.ts]: Remove the `.` at the end.

**Comment by yuit** [comment]
@maylukas Thank you for the PR!, could you fix the standing build break and I will merge the PR in once it is green :)

**Comment by maylukas** [comment]
@yuit I think I fixed the error but Travis gives me the error "The job exceeded the maximum time limit for jobs, and has been terminated.". Is there any way to extend the time limit?

**Comment by andy-ms** [comment]
Ran tests manually.

**Comment by maylukas** [comment]
@andy-ms Thanks!
